File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    answer = []
    progress_days=[math.ceil((100-x)/speeds[idx]) for idx,x in enumerate(progresses)]

    queue=deque(progress_days)

    done=queue[0]
    count=0

    while queue:
        if queue[0]<=done: #9<=9
            # 빠져나가도됌
            queue.popleft()
            count+=1 #2 #1
        else:
            answer.append(count) #[2]
            count = 0
            done=queue[0]
    answer.append(count)


    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
  
    //1. 작업일 배열 생성
    const work=progresses.map((v,i)=>Math.ceil((100-v)/speeds[i]));
    //2. 순회

    let deploy=work[0];
    let count=0;
    
    for(let i=0;i<work.length;i++){
        
        if(work[i]>deploy){
            answer.push(count);
            count=0;
            deploy=work[i];
        }
        count++;
    }
    answer.push(count); //마지막 요소 push
    return answer;

}